lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -733,13 +764,19 @@ def generate(cls, _xmodule_instance_args, _entry_id, course_id, task_input, acti
         task_progress = TaskProgress(action_name, num_reports, start_time)
         current_step = {'step': 'Calculating students answers to problem'}
         task_progress.update_task_state(extra_meta=current_step)
-        problem_location = task_input.get('problem_location')
+        problem_locations = task_input.get('problem_locations')
```

Comment:
You're making a backwards incompatible change in the arguments here, for something that's persisted in the database. Please change the routing key for the task that eventually triggers this so that we don't end up in a weird state where new code is trying to parse old data or vice versa in the case of a rollback.

Reply:
I'm not entirely sure how to do that but am looking into it. 

Alternatively I could also (as a temporary backwards-compatibility workaround) have the new code try to get `problem_location` if `problem_locations` is not present. It can also save the first of the problem locations into `problem_location` so that if a new task is processed by old code it will still work. I don't think there will be any multi-problem reports in the time between a deployment and rollback since this change is currently API-only.

Reply:
Come to think of it, a separate routing key probably isn't necessary. I think it's sufficient if you put an explicit `name=` attribute on the `@task` decorator for the launching celery task, and set it to the something like `@task(name=lms.djangoapps.instructor_task.tasks.calculate_problem_grade_report.v2)`. The old messages go to whatever name was dynamically generated, and the new messages go to this name. (@jmbowman: May I ask you to sanity check my understanding of celery routing and task versioning?)

Reply:
Thanks! I have made this change.